src/backend/tests/Fusion.Resources.Functions.Tests/Notifications/ScheduledReportNotificationBuilderTests.cs

```diff
@@ -0,0 +1,246 @@
+                OrgPositionInstance = new ApiPositionInstanceV2 { AppliesFrom = DateTime.UtcNow.AddMonths(4) },
+            },
+            new()
+            {
+                Type = "SomethingElse",
+                State = "created",
+                OrgPositionInstance = new ApiPositionInstanceV2 { AppliesFrom = DateTime.UtcNow.AddMonths(2) },
+            },
+            new()
+            {
+                Type = "ResourceOwnerChange",
+                State = "completed",
+                OrgPositionInstance = new ApiPositionInstanceV2 { AppliesFrom = DateTime.UtcNow.AddMonths(4) },
+            },
+        };
+
+
+        // Act
+        var numberOfRequests =
+            ResourceOwnerReportDataCreator.GetNumberOfRequestsStartingInMoreThanThreeMonths(requests);
+
+        // Assert
+        numberOfRequests.Should().Be(1);
+    }
+
+    [Fact]
+    public void GetTotalNumberOfPersonnel_ShouldReturnCorrectNumberOfPersonnel()
+    {
+        // Arrange
+        var personnel = NotificationReportApiResponseMock.GetMockedInternalPersonnel(5, 100, 0, 0, 0);
+
+        // Act
+        var totalNumberOfPersonnel = ResourceOwnerReportDataCreator.GetTotalNumberOfPersonnel(personnel);
+
+        // Assert
+        totalNumberOfPersonnel.Should().Be(5);
+    }
+
+    [Fact]
+    public void GetNumberOfRequestsStartingInLessThanThreeMonths_ShouldReturnCorrectNumberOfRequests()
+    {
+        // Arrange
+        var requests = new List<IResourcesApiClient.ResourceAllocationRequest>
+        {
+            // Will pass
+            new()
+            {
+                Type = "SomethingElse",
+                State = "created",
+                OrgPositionInstance = new ApiPositionInstanceV2 { AppliesFrom = DateTime.UtcNow.AddMonths(2) },
+            },
+            new()
+            {
+                Type = "SomethingElse",
+                State = "created",
+                OrgPositionInstance = new ApiPositionInstanceV2 { AppliesFrom = DateTime.UtcNow.AddMonths(4) },
+            },
+            new()
+            {
+                Type = "ResourceOwnerChange",
+                State = "completed",
+                OrgPositionInstance = new ApiPositionInstanceV2 { AppliesFrom = DateTime.UtcNow.AddMonths(2) },
+            },
+        };
+
+        // Act
+        var numberOfRequests =
+            ResourceOwnerReportDataCreator.GetNumberOfRequestsStartingInLessThanThreeMonths(requests);
+
+        // Assert
+        numberOfRequests.Should().Be(1);
+    }
+
+    [Fact]
+    public void GetAllocationChangesAwaitingTaskOwnerAction_ShouldReturnCorrectNumberOfRequests()
+    {
+        // Arrange
+        var requests = new List<IResourcesApiClient.ResourceAllocationRequest>
+        {
+            // Will pass
+            new()
+            {
+                Type = "ResourceOwnerChange",
+                State = "created",
+            },
+            new()
+            {
+                Type = "SomethingElse",
```

Comment:
Maybe explicit define what other types that are possible here? This makes the tests even more self-explanatory.
This is related to other places where we defined types also.